bpdm-pool/src/main/kotlin/org/eclipse/tractusx/bpdm/pool/service/RequestValidationService.kt

```diff
@@ -393,87 +410,102 @@ class RequestValidationService(
-                    ErrorInfo(error, "Duplicate Address Identifier: Value '${it.value}' of type '${it.type}'", entityKey.getRequestKey())
-                else
-                    null
+    private inner class ValidateAddressIdentifiersDuplicated<ERROR : ErrorCode>(
+        addresses: Collection<IBaseLogisticAddressDto>,
+        private val error: ERROR
+    ) {
+
+        val existingIdentifiers = getAddressDuplicateIdentifierCandidates(addresses)
+
+        fun validate(address: IBaseLogisticAddressDto, entityKey: RequestWithKey, bpn: String?): Collection<ErrorInfo<ERROR>> {
+
+            return address.identifiers.mapNotNull {
+                val identifierPair = IdentifierCandidateKey(type = it.type, value = it.value)
+                existingIdentifiers[identifierPair]?.let { candidate ->
+                    if (candidate.bpn === null || candidate.bpn != bpn)
+                        ErrorInfo(error, "Duplicate Address Identifier: Value '${it.value}' of type '${it.type}'", entityKey.getRequestKey())
+                    else
+                        null
+                }
             }
         }
+
+        private fun getAddressDuplicateIdentifierCandidates(requests: Collection<IBaseLogisticAddressDto>)
+                : Map<IdentifierCandidateKey, IdentifierCandidate> {
+            val identifiers = requests.flatMap { it.identifiers }
+            val idValues = identifiers.map { it.value }
+            val duplicatesFromRequest = identifiers
+                .map { IdentifierCandidateKey(it.type, it.value) }
+                .findDuplicates()
+                .associateWith { IdentifierCandidate(bpn = null, type = it.type, value = it.value) }
+            val duplicatesFromDb = addressIdentifierRepository.findByValueIn(idValues)
+                .map { IdentifierCandidate(bpn = it.address.bpn, type = it.type.technicalKey, value = it.value) }
+                .associateBy { IdentifierCandidateKey(it.type, it.value) }
+            return duplicatesFromRequest.plus(duplicatesFromDb)
+        }
     }
 
-    private fun validateAddressParent(
-        request: AddressPartnerCreateRequest,
-        type: BusinessPartnerType?,
-        existingLegalEntities: Set<String>,
-        existingSites: Set<String>
-    )
-            : Collection<ErrorInfo<AddressCreateError>> {
-        return when (type) {
-            BusinessPartnerType.LEGAL_ENTITY -> validateParentBpnExists(
-                request.bpnParent,
-                request.index,
-                existingLegalEntities,
-                AddressCreateError.LegalEntityNotFound
-            )
+    private inner class ValidateAddressParent(requests: Collection<AddressPartnerCreateRequest>) {
+
+        val requestsWithParentType = requests.map { Pair(it, bpnIssuingService.translateToBusinessPartnerType(it.bpnParent)) }
+        val requestsByParentType = requestsWithParentType.groupBy({ it.second }, { it.first })
```

Comment:
You could simplify: 
```suggestion
      val requestsByParentType = requests.groupBy { bpnIssuingService.translateToBusinessPartnerType(it.bpnParent) }
```

Reply:
this is also already done in the next step